template.go

```diff
@@ -159,3 +159,26 @@ func (gc *GatewayClient) GetTemplateByFilters(key string, value string) ([]types
 
 	return templates.TemplateDetails, nil
 }
+
+// CloneTemplate Creates a new Template based on a preexisting Template using the original template id
+func (gc *GatewayClient) CloneTemplate(s *System, originTemplateID string, templateName string) error {
+	defer TimeSpent("CloneTemplate", time.Now())
+	path := `/Api/V1/ServiceTemplate/cloneTemplate`
+
+	template, err := gc.GetTemplateByFilters("originalTemplateId", originTemplateID)
+	if err != nil {
+		return fmt.Errorf("Error While Cloning Template: %s", err.Error())
+	}
+
+	template[0].TemplateLocked = false
```

Comment:
Will `template` always at least length 1?

Reply:
yes, the return is a list since the GetTemplateByFilters will never know how many would be returned, if it isn't retrieved based on the filter provided it'll error out

Reply:
Could the error be nil and an empty slice returned? If so, we should check the length before indexing it.

Reply:
on line 156 it has that check you mentioned, so checking the length won't be necessary. if it reaches past the GetTemplateTemplateByFilters call on line 168 it will have at least 1. I think my comment was a little confusing earlier